test/dom.cpp

```diff
@@ -23,30 +23,108 @@
  */
 #include <migraphx/dom_info.hpp>
 #include <migraphx/program.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/iterator_for.hpp>
 #include <basic_ops.hpp>
 #include <test.hpp>
 
+bool dominates_self(const migraphx::dominator_info& dom, const migraphx::module& m)
+{
+    return migraphx::any_of(migraphx::iterator_for(m), [&](auto ins) {
+        return dom.strictly_dominate(ins, ins);
```

Comment:
This would always return false.  https://github.com/ROCm/AMDMIGraphX/blob/cc7c280162e0fcf9e08dce71f8abe6cede1089a4/src/dom_info.cpp#L37

Reply:
Yes it should always return false, which is why I added a check for that